tools/cpp/src/cpp-build/generate_geocoding_data.cc

```diff
@@ -102,20 +102,21 @@ bool ListDirectory(const string& path, vector<DirEntry>* entries) {
     return false;
   }
   AutoCloser<DIR> dir_closer(&dir, closedir);
-  struct dirent entry, *dir_result;
+  struct dirent *entry;
   struct stat entry_stat;
   while (true) {
-    const int res = readdir_r(dir, &entry, &dir_result);
-    if (res) {
+    errno = 0;
+    entry = readdir(dir);
+    if (errno) {
```

Comment:
The standard does actually not guarantee that a call to readdir() that returns a non-NULL pointer doesn't also modify errno, that guarantee is only specified for a call that returns NULL. So checking errno before checking the return value could possibly yield a false positive.

A safer implementation would be:

if (entry == NULL) {
  return errno == 0;
}